src/api/instance.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * HTTP request용 모듈
+ *
+ * @async
+ * @function
+ * @param {string} url 요청 URL
+ * @param {string} method 요청할 HTTP 메서드
+ * @param {Object} [body] 요청 본문 데이터(optional)
+ * @param {Object} [headers] 요청에 추가적인 헤더(optional)
+ * @returns {Promise} 응답 데이터
+ * @throws {Error} 응답 상태가 'ok'가 아닐 경우 에러발생
+ */
+
+const baseUrl = 'https://bootcamp-api.codeit.kr/api';
+
+const request = async (url, method, body, headers = {}) => {
+  const options = {
+    method,
+    headers: {
+      'Content-Type': 'application/json',
+      ...headers,
+    },
+    body: body ? JSON.stringify(body) : undefined,
+  };
+
+  const res = await fetch(baseUrl + url, options);
+  const data = await res.json();
+
+  if (res.ok) return data;
+  else throw Error(data);
+};
```

Comment:
> 3. 에러객체의 메시지를 활용하면 좋겠는데 콘솔에 찍어보니 instance.js에선 아래처럼 나오는데 authAPI.js로 넘어오면 이렇게 되더라구요. 보통 이런 에러케이스들은 어떤식으로 처리하는게 맞나요?

만약 이어서 받고 싶으면, 아래처럼 작성해주시면 됩니다.
Error로 감싸주게 되면서 에러객체가 또 다시 에러객체로 감싸졌다고 생각하시면 될 것 같아요
https://developer.mozilla.org/ko/docs/Web/JavaScript/Reference/Global_Objects/Error


```suggestion
  if (res.ok) return data;
  else throw data;
};
```

실무에서는 sentry를 통해, FE 에러들을 로깅하는 편이에요. 근데 커스텀하게 객체를 가져가진 않는 것 같습니다.